src/ert/ensemble_evaluator/evaluator.py

```diff
@@ -139,27 +122,31 @@ def set_event_handler(event_types: set[type[Event]], func: Any) -> None:
         set_event_handler({EnsembleFailed}, self._failed_handler)
 
         while True:
-            batch: list[tuple[EVENT_HANDLER, Event]] = []
+            function_to_events_map: dict[EVENT_HANDLER, list[Event]] = defaultdict(list)
+            events_in_batch_count = 0
             start_time = asyncio.get_running_loop().time()
             while (
-                len(batch) < self._max_batch_size
+                events_in_batch_count < self._max_batch_size
                 and asyncio.get_running_loop().time() - start_time
                 < self._batching_interval
             ):
-                self._complete_batch.clear()
                 try:
                     event = self._events.get_nowait()
+                    events_in_batch_count += 1
```

Comment:
What I still question and is not clear to me is that this entire events batching sleeps while processing of events happens. This when having as two separate tasks can batch events in parallel asynchronously.